api4/post.go

```diff
@@ -428,3 +430,25 @@ func getFileInfosForPost(c *Context, w http.ResponseWriter, r *http.Request) {
 		w.Write([]byte(model.FileInfosToJson(infos)))
 	}
 }
+
+func doPostAction(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequirePostId()
+	if c.Err != nil {
+		return
+	}
+
+	if !app.SessionHasPermissionToChannelByPost(c.Session, c.Params.PostId, model.PERMISSION_READ_CHANNEL) {
+		c.SetPermissionError(model.PERMISSION_READ_CHANNEL)
+		return
+	}
+
+	params := mux.Vars(r)
+	id := params["id"]
```

Comment:
Can you add this to `api4/params.go` and add a context function `RequireActionId()`? Then you can just do `c.RequirePostId().RequireActionId()` at the top here and for any future functions that need an action id